### ISSUE TITLE
[bug] Self diagnostics doesn't create logs file

### ISSUE THREAD

**Issue by lcssanches** [open-telemetry/opentelemetry-dotnet#5992]
### Package

OpenTelemetry

### Package Version

| Package Name                          | Version |
| ------------------------------------- | ------- |
| OpenTelemetry | 1.10.0 |
| OpenTelemetry.Api | 1.10.0 |
| OpenTelemetry.Api.ProviderBuilderExtensions | 1.10.0 |
| OpenTelemetry.Exporter.Console | 1.10.0 |
| OpenTelemetry.Instrumentation.AspNet | 1.9.0-beta.1 |
| OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule | 1.9.0-beta.1 |
| OpenTelemetry.Instrumentation.Http | 1.9.0 |
| OpenTelemetry.Instrumentation.SqlClient | 1.9.0-beta.1 |
| OpenTelemetry.PersistentStorage.Abstractions | 1.0.0 |
| OpenTelemetry.PersistentStorage.FileSystem | 1.0.0 |

### Runtime Version

net472

### Description

I followed this self [diagnostic guide](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#self-diagnostics), but the log file is never created. 

This is the `OTEL_DIAGNOSTICS.json` configuration file created:
```json
{
  "LogDirectory": ".",
  "FileSize": 32768,
  "LogLevel": "Verbose"
}
```

The only endpoint in this API is checking for the mentioned directories where it would look for the diagnostics file, and it exists.
```json
{
    "AppContext_BaseDirectory": "C:\\Users\\lcssa\\source\\repos\\TestDiagnostics\\TestDiagnostics\\",
    "AppContext_BaseDirectory_FileExists": true,
    "Directory_GetCurrentDirectory": "C:\\Program Files\\IIS Express",
    "Directory_GetCurrentDirectory_FileExists": false,
    "FileExists": false
}
```

List of files in that directory
```
    Directory: C:\Users\lcssa\source\repos\TestDiagnostics\TestDiagnostics

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----          2024-11-19    18:19                App_Data
d----          2024-11-19    18:44                App_Start
d----          2024-11-19    19:07                bin
d----          2024-11-19    18:54                Controllers
d----          2024-11-19    19:07                ILLink
d----          2024-11-19    18:19                Models
d----          2024-11-19    18:19                obj
d----          2024-11-19    18:19                Properties
-a---          2024-11-19    18:19          32038 favicon.ico
-a---          2024-11-19    18:19            110 Global.asax
-a---          2024-11-19    19:07           2612 Global.asax.cs
-a---          2024-11-19    19:01             75 OTEL_DIAGNOSTICS.json
-a---          2024-11-19    19:07           4925 packages.config
-a---          2024-11-19    19:07          21936 TestDiagnostics.csproj
-a---          2024-11-19    18:49           1469 TestDiagnostics.csproj.user
-a---          2024-11-19    19:07           7667 Web.config
-a---          2024-11-19    18:19           1272 Web.Debug.config
-a---          2024-11-19    18:19           1333 Web.Release.config
```

### Steps to Reproduce

Run the project: https://github.com/[lcssanches/otel-diagnostics-example](https://github.com/lcssanches/otel-diagnostics-example)

### Expected Result

File logs are created, as mentioned [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#self-diagnostics).

### Actual Result

No file logs are created.

**Comment by lcssanches** [comment]
Found the issue.
TLDR; If you are running IIS, specify the absolute path in `LogDirectory`.
